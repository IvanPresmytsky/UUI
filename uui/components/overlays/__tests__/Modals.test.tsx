import React from 'react';
import {
    ModalBlocker, ModalHeader, ModalFooter, ModalWindow,
} from '../Modals';
import { renderSnapshotWithContextAsync } from '@epam/uui-test-utils';

describe('Modals', () => {
    it('should be rendered correctly with minimum props', async () => {
        const tree = await renderSnapshotWithContextAsync(
            <ModalBlocker key="blocker" isActive zIndex={ 1 } abort={ jest.fn } success={ jest.fn }>
                <ModalWindow>
                    <ModalHeader />
                    <ModalFooter />
                </ModalWindow>
            </ModalBlocker>,
        );
        expect(tree).toMatchSnapshot();
    });

    it('should be rendered correctly', async () => {
        const tree = await renderSnapshotWithContextAsync(
            <ModalBlocker key="blocker" isActive zIndex={ 1 } abort={ jest.fn } success={ jest.fn } disallowClickOutside>
                <ModalWindow height={ 300 } width={ 300 } onClick={ jest.fn }>
                    <ModalHeader title="Test header" onClose={ jest.fn } borderBottom margin="12" size="36" spacing="6" padding="6" topShadow vPadding="12" />
                    <ModalFooter borderBottom margin="12" size="48" spacing="12" padding="18" topShadow vPadding="24" />
                </ModalWindow>
            </ModalBlocker>,
        );
        expect(tree).toMatchSnapshot();
    });
});
