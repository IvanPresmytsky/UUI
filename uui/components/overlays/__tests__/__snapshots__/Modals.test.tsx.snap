// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modals should be rendered correctly with many props 1`] = `
<div
  className="container modalBlocker"
  style={
    Object {
      "zIndex": 1,
    }
  }
>
  <div
    aria-label="Click to close a modal"
    className="uui-modal-blocker"
    onClick={[Function]}
  />
  <div
    aria-modal={true}
    className="uui-modal-window root modal container"
    role="dialog"
    style={
      Object {
        "height": 300,
        "maxHeight": "80dvh",
        "width": 300,
      }
    }
  >
    <div
      className="uui-flex-row root uui-size-36 padding-6 vPadding-12 borderBottom root modalHeader container align-items-center"
      style={
        Object {
          "columnGap": "12px",
          "rowGap": undefined,
        }
      }
    >
      <div
        className="root uui-text uui-size-48 lineHeight fontSize uui-color-primary uui-font-weight-600 uui-font-style-normal uui-typography container"
        style={
          Object {
            "--uui-text-font-size": "18px",
          }
        }
      >
        Test header
      </div>
      <div
        className="flexSpacer"
      />
      <div
        className="flexCell"
        style={
          Object {
            "alignSelf": undefined,
            "flexBasis": "auto",
            "flexGrow": undefined,
            "flexShrink": 0,
            "minWidth": 0,
            "textAlign": undefined,
          }
        }
      >
        <button
          aria-label="Close modal"
          className="uui-button-box uui-enabled -clickable container uui-icon_button uui-color-neutral root"
          onClick={[Function]}
          tabIndex={0}
          type="button"
        >
          <div
            className="container uui-icon uui-enabled"
            style={Object {}}
          >
            <svg
              className=""
            />
          </div>
        </button>
      </div>
    </div>
    <div
      className="uui-flex-row root uui-size-36 padding-18 vPadding-24 root modalFooter container align-items-center"
      style={
        Object {
          "columnGap": "12px",
          "rowGap": undefined,
        }
      }
    />
  </div>
</div>
`;

exports[`Modals should be rendered correctly with minimum props 1`] = `
<div
  className="container modalBlocker"
  style={
    Object {
      "zIndex": 1,
    }
  }
>
  <div
    aria-label="Click to close a modal"
    className="uui-modal-blocker"
    onClick={[Function]}
  />
  <div
    aria-modal={true}
    className="uui-modal-window root modal container"
    role="dialog"
    style={
      Object {
        "height": "auto",
        "maxHeight": "80dvh",
        "width": "420",
      }
    }
  >
    <div
      className="uui-flex-row root uui-size-36 padding-24 vPadding-12 root modalHeader container align-items-center"
      style={
        Object {
          "columnGap": "12px",
          "rowGap": undefined,
        }
      }
    />
    <div
      className="uui-flex-row root uui-size-36 padding-24 vPadding-24 root modalFooter container align-items-center"
      style={
        Object {
          "columnGap": "12px",
          "rowGap": undefined,
        }
      }
    />
  </div>
</div>
`;
